stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,272 @@ def dispatch(self, **kwargs):
                                  Payload=alert_string)
 
         return self._log_status(resp)
+
+@output
+class JiraOutput(StreamOutputBase):
+    """JiraOutput handles all alert dispatching for Jira"""
+    __service__ = 'jira'
+
+    LOGIN = 'rest/auth/1/session'
+    SEARCH = 'rest/api/2/search'
+    ISSUE = 'rest/api/2/issue'
+    COMMENT = 'rest/api/2/issue/{}/comment'
+
+    def __init__(self, *args, **kwargs):
+        StreamOutputBase.__init__(self, *args, **kwargs)
+        self.base_url = None
+        self.auth_cookie = None
+
+    def get_user_defined_properties(self):
+        """Get properties that must be asssigned by the user when configuring a new Jira
+        output.  This should be sensitive or unique information for this use-case that needs
+        to come from the user.
+
+        Every output should return a dict that contains a 'descriptor' with a description of the
+        integration being configured.
+
+        Jira requires a username, password, URL, project key, and issue type for alert dispatching.
+        These values should be masked during input and are credential requirements.
+
+        An additional parameter 'aggregate' is used to determine if alerts are aggregated into a
+        single Jira issue based on the StreamAlert rule.
+
+        Returns:
+            OrderedDict: Contains various OutputProperty items
+        """
+        return OrderedDict([
+            ('descriptor',
+             OutputProperty(description='a short and unique descriptor for this '
+                                        'Jira integration')),
+            ('username',
+             OutputProperty(description='the Jira username',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('password',
+             OutputProperty(description='the Jira password',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('url',
+             OutputProperty(description='the Jira url',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('project_key',
+             OutputProperty(description='the Jira project key',
+                            mask_input=False,
+                            cred_requirement=True)),
+            ('issue_type',
+             OutputProperty(description='the Jira issue type',
+                            mask_input=False,
+                            cred_requirement=True)),
+            ('aggregate',
+             OutputProperty(description='the Jira aggregation behavior to aggregate '
```

Comment:
@ryandeivert I was meaning to ask you about this.  The question is asked to the user in a very awkward way, due to "Please provide" being prefixed to these descriptions.  I essentially want to ask "Do you want to aggregate alerts?".  In its current form, the user is asked "Please provide the aggregation behavior to aggregate alerts by rule name (yes/no)", which is quite the mouthful and I'm not entirely sure if it makes any sense.  Let me know if you have any suggestions.

Reply:
lol yeah that is a bit corny. we may have to just go with it for now and address it in the next alert processor refactor

Reply:
It'll probably be best to just have the full prompt message here (and change the `OutputProperty`'s `description` attribute to something better, like `prompt`)

Reply:
Just to clarify -  wasn't implying you should make that change :) I'll handle it in the future